### ISSUE TITLE
Refine BatchNormalization

### ISSUE THREAD

**Issue by yiheng** [intel-analytics/BigDL#1599]
1. Use weight/bias to store runningMean/runningVar. The coresponding gradWeight/gradBias always are zero.
2. When batch size is small, consider to adjust saveMean/saveVar with runningMean/runningVar

**Comment by yiheng** [comment]
There's a problem in the first solution, when the weight decay is enabled. Now the runningMean/runningVar has been stored in extra parameters of the model.

The second solution is similiar with batch renorm. Close this issue and create a new one.
